packages/grid/x-data-grid/src/components/GridRow.tsx

```diff
@@ -248,14 +250,28 @@ function GridRow(props: React.HTMLAttributes<HTMLDivElement> & GridRowProps) {
     [apiRef, onClick, publish, rowId],
   );
 
+  const sizes = apiRef.current.unstable_getRowInternalSizes(rowId);
+
+  let minHeight = rowHeight;
+  if (minHeight === 'auto' && sizes && Object.keys(sizes).length > 1) {
```

Comment:
> Object.keys(sizes).length > 1

Otherwise it means there's no pinned columns and we can set `minHeight` to `auto`?